File: request-manager/src/main/java/fii/workflow/manager/repository/GroupRepository.java
package fii.workflow.manager.repository;

import fii.workflow.manager.domain.Customer;
import fii.workflow.manager.domain.CustomerGroup;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Modifying;
import org.springframework.data.jpa.repository.Query;
import org.springframework.transaction.annotation.Transactional;

import java.util.Set;

public interface GroupRepository extends JpaRepository<CustomerGroup, Long> {
    Set<CustomerGroup> findByCustomers(Customer c);

    @Modifying
    @Transactional
    @Query(value = "DELETE from customer_to_group where customer_id=?1 and group_id=?2", nativeQuery = true)
    void deleteMemberByCustomerIdAndGroupId(Long customerId, Long groupId);
}

File: workflow-manager-frontend/app/src/main/java/com/example/workflow_manager_frontend/data/repository/GroupRepository.kt
package com.example.workflow_manager_frontend.data.repository

import com.example.workflow_manager_frontend.domain.Customer
import com.example.workflow_manager_frontend.domain.Group

interface GroupRepository {
    suspend fun getGroupsByCustomerId(customerId : Int) : List<Group>?

    suspend fun getGroupById(groupId: Int) : Group?

    suspend fun addCustomerToGroup(groupId: Int, customer: Customer)

    suspend fun deleteCustomerFromGroup(groupId: Int, customerId: Int)
}